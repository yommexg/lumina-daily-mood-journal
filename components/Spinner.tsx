import { View, ActivityIndicator, StyleSheet } from "react-native";

const Spinner = () => {
  return (
    <View style={styles.container}>
      <View style={styles.overlay} />
      <ActivityIndicator
        size="large"
        color="#fff"
      />
    </View>
  );
};

export default Spinner;

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    zIndex: 50,
    width: "100%",
    height: "100%",
    justifyContent: "center",
    alignItems: "center",
  },
  overlay: {
    position: "absolute",
    width: "100%",
    height: "100%",
    backgroundColor: "black",
    opacity: 0.4,
  },
});
