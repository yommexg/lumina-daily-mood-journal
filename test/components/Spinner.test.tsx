import Spinner from "@/components/Spinner";
import { render } from "@testing-library/react-native";
import React from "react";

describe("Spinner", () => {
  it("renders an ActivityIndicator inside a styled overlay container", () => {
    const { getByTestId, toJSON } = render(<Spinner />);

    // Find the ActivityIndicator by its type
    const activityIndicator = getByTestId("spinner-activity-indicator");
    expect(activityIndicator).toBeTruthy();

    // Snapshot test to validate structure and styles
    expect(toJSON()).toMatchSnapshot();
  });
});
