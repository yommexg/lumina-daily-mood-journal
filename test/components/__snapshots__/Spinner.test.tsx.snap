// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders an ActivityIndicator inside a styled overlay container 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "100%",
      "zIndex": 50,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "black",
        "height": "100%",
        "opacity": 0.4,
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
    testID="spinner-activity-indicator"
  />
</View>
`;
